code/modules/antagonists/ninja/energy_katana.dm

```diff
@@ -78,3 +81,13 @@
 
 /datum/action/innate/dash/ninja/HideFrom(mob/viewer)
 	return
+
+/// Teleports to a tile adjacent to a mob, then attacks it
+/datum/action/innate/dash/ninja/proc/attack_teleport(mob/living/user, mob/living/stabbing)
+	var/list/turf/line = get_line(user, stabbing)
+	var/obj/item/sword = target
+	if(!teleport(user, line[length(line) - 1])) // teleports to the second last turf, should be adjacent to the target
```

Comment:
This could theoretically runtime (in a very cursed scenario) if the user is inside of a bluespace bodybag inside/on the same tile as the target? Or ends up failing CanReach while being on the same tile in some other way, maybe check that line is at least three tiles in length? (To avoid same-tile teleports as well)